clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -659,25 +623,25 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                                         keyByteArray == null ? ConsumerRecord.NULL_SIZE : keyByteArray.length,
                                         valueByteArray == null ? ConsumerRecord.NULL_SIZE : valueByteArray.length,
                                         key, value);
-        } catch (KafkaException e) {
+        } catch (InvalidRecordException e) {
             throw e;
```

Comment:
The other thing is that the `InvalidRecordException` is a bit scarce in details. It would be useful to at least include the partition and offset like we do for the `SerializationException`.